libraries/lib-time-and-pitch/CMakeLists.txt

```diff
@@ -4,6 +4,8 @@ Provides time-stretching and pitch-shifting functionality.
 Based on the more versatile time-stretching from StaffPad, but tailored for the needs of Audacity.
 ]]
 
+add_definitions( -DCMAKE_CURRENT_SOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}" )
```

Comment:
This page recommends `add_compile_definitions` instead https://cmake.org/cmake/help/latest/command/add_definitions.html

You use this macro in `FormantShifterLogger.cpp`.  I would choose a different name for the macro, like THIS_DIRECTORY, just so we aren't confusing compile-time preprocessor symbols with Cmake configure-time variables.

This commit fails to compile.

```
/Users/paullicameli/Github/audacity/libraries/lib-time-and-pitch/FormantShifter.cpp:104:31: error: no member named 'getSize' in 'staffpad::audio::FourierTransform'
   const auto fftSize = mFft->getSize();
                        ~~~~~~^
```